include/zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt_client.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_IMG_MGMT_CLIENT_
+#define H_IMG_MGMT_CLIENT_
+
+#include <inttypes.h>
+#include <zephyr/mgmt/mcumgr/grp/img_mgmt/img_mgmt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Image list data.
+ */
+struct mcumgr_image_data {
+	int slot_num;
+	int img_num;
+	char hash[IMG_MGMT_HASH_LEN];
+	char version[IMG_MGMT_VER_MAX_STR_LEN + 1];
+	bool bootable : 1;
+	bool pending : 1;
+	bool confirmed : 1;
+	bool active : 1;
+	bool permanent : 1;
+};
+
+/**
+ * @brief MCUmgr Image list response.
+ *
+ *	Structure image_list_length and image_list pointer are defined by user for store received
+ *	image list data. image_list_length is updated by MCUmgr client for response.
+ */
+struct mcumgr_image_state {
+	/** status */
+	int status;
+	/** Length of image_list buffer */
+	int image_list_length;
+	/** Image list pointer */
+	struct mcumgr_image_data *image_list;
+};
+
+/**
+ * @brief MCUmgr Image upload response.
+ */
+struct mcumgr_image_upload {
+	/** status */
+	int status;
+	/** Reported image offset */
+	size_t image_upload_offset;
+};
+
+/**
+ * @brief Init started Image Upload Process
+ *
+ * @param image_size Size of image in bytes.
+ *
+ * @param image_num Image Num.
+ *
+ * @param image_hash Pointer to HASH for image. Use NULL when HASH from image is not available.
+ *
+ * @return 0 on success, #mcumgr_err_t code on failure.
+ */
+int img_mgmt_client_upload_init(size_t image_size, int image_num, const char *image_hash);
+
+/**
+ * @brief Upload part of image
+ *
+ * @param data Pointer to data.
+ *
+ * @param length Length of data
+ *
+ * @param res_buf Pointer for command response structure.
+ *
+ * @return 0 on success, #mcumgr_err_t code on failure.
+ */
+int img_mgmt_client_upload(const uint8_t *data, size_t length, struct mcumgr_image_upload *res_buf);
+
+/**
+ * @brief Image state set for Image Confirm and test
+ *
+ * @param hash		Pointer to Hash (Needed for test).
+ *
+ * @param hash_len	length of Hash (Needed for test).
```

Comment:
Not sure we need a hash_len param, either the hash is not provided (hash == NULL) or it should be the size of the define